src/lib/utils/tracing/systemtap.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <sys/sdt.h>
+
+// Systemtap is not available on OS X. The alternative would be dtrace. Actually, systemtap probes are compatible
+// with dtrace probes and the other way round. However, Apple's dtrace implementation differs from the original solaris
+// implementation. Thus, Apple's dtrace does not understand the USDT probe definition. In order to be able to compile
+// the project on Mac we define the probes in this file.
+
+// In order to be compatible with Mac dtrace probes, both provider and probe name have to be uppercase
+constexpr bool is_valid_name(const char* name) {
+  do {
+    if (*name >= 'a' && *name <= 'z') return false;
+    if (*name == '-') return false;
+  } while (*++name != '\0');
+  return true;
+}
+
+#if defined(__APPLE__) || defined(__MACOS__)
```

Comment:
Why do you check for apple in a file called "systemtap"? Maybe replace with a generic name?